lib/clustering/mean_shift.test.cc

```diff
@@ -1,44 +1,107 @@
 /// \file
```

Comment:
Interesting, I never realized, how hard it is to verify mean shift via unit tests. It's so easy visually. I think we leave it like that for the moment.